docs/contributor/adr/004-layered-architecture.md

```diff
@@ -0,0 +1,77 @@
+# ADR 004 - Adoption of Layered Architecture (3-Tier)
+
+## Status
+
+Accepted
+
+## Context
+
+We need to design a maintainable, testable, and scalable architecture for lifecycle-manager, which interacts with Kubernetes APIs and includes business logic and orchestration of resources.
+
+To achieve this, we have adopted a Layered Architecture pattern with the following three layers:
+1. Controller Layer (Top)
+2. Service Layer (Middle)
+3. Repository Layer (Bottom) 
+
+Each layer has distinct responsibilities and clear dependency rules to ensure separation of concerns and minimize coupling.
+
+## Decision
+
+### Controller Layer
+Responsibility:
+- Serves as the orchestrator for reconciling Kubernetes resources
+- Handles event processing and decision-making around requeue intervals and error handling
+
+Key Points:
+- Depends on Service Layer
+- Uses the results from service calls to make decisions about reconciliation behavior
+- It is the only layer that creates and manages controller-runtime’s ctrl.Result objects
+- Does not perform business logic directly; delegates all such work to services
+
+### Service Layer
+Responsibility:
+- Implements business logic by orchestrating one or more repositories
+- Encapsulates complex workflows and business rules
+
+Key Points:
+- Depends on Repository Layer
+- Must not access Kubernetes APIs directly; instead, uses repositories for any data access
+- Contains pure business logic to ensure it can be tested independently of Kubernetes
+
+### Repository Layer
+Responsibility:
+- Acts as the CRUD (Create, Read, Update, Delete) data layer
+- Directly interacts with the Kubernetes API using the controller-runtime client
+- Responsible for all data persistence and retrieval operations
```

Comment:
I would mention this layer should not contain any business logic or decision-making, as a clear guide for this case https://github.com/kyma-project/lifecycle-manager/pull/2453/files#r2079172656